src/bin/pg_basebackup/t/050_check_recovery_backup.pl

```diff
@@ -0,0 +1,122 @@
+
+	print("Running: restore_backup_primary: '$restore_point' '$label'\n");
+
+	$primary->command_ok([ 'rm', '-rf', $primary_pgdata ],
+		"'$label': rm primary pgdata");
+	$primary->command_ok([ 'cp', '-r', $backup_dir, $primary_pgdata ],
+		"'$label': copy basebackup to primary");
+
+	# configure recovery
+	$primary->set_recovery_mode;
+	open my $conf, '>', "$primary_pgdata/postgresql.auto.conf";
+	print $conf "recovery_target_action = 'promote'\n";
+	print $conf "restore_command = 'cp -vr $archivedir/%f $primary_pgdata/%p'\n";
+	print $conf "recovery_target_name = '$restore_point'\n";
+	close $conf;
+	$primary->start;
+}
+
+sub wait_recovery_and_switch_wal_primary
+{
+	my ($label) = @_;
+
+	print("Running: wait_recovery_and_switch_wal_primary: '$label'\n");
+
+	$primary->poll_query_until('test_db', "SELECT true FROM pg_switch_wal()")
+		or die "'$label': Timed out while waiting for switch WAL after recovery";
+}
+
+mkdir($archivedir);
+
+# Initialize primary
+$primary->init(allows_streaming => 1, extra => ['--data-checksums']);
+$primary->append_conf("postgresql.conf", "archive_mode = 'on'");
+$primary->append_conf("postgresql.conf", "archive_command = 'cp -rv %p $archivedir/'");
+$primary->start;
+$primary->safe_psql('postgres', 'CREATE DATABASE test_db');
+$primary->safe_psql('test_db', 'CREATE TABLE test as select generate_series(1,10)');
+$primary->safe_psql('postgres', 'CREATE ROLE postgres WITH LOGIN REPLICATION');
+
+
+create_backup_primary($basebackupdir1, 'backup_label1');
+$primary->stop('smart');
+restore_backup_primary($basebackupdir1, 'backup_label1', '001 -> 012');
+wait_recovery_and_switch_wal_primary('001 -> 012');
+
+create_backup_primary($basebackupdir2, 'backup_label2');
+$primary->stop('smart');
+restore_backup_primary($basebackupdir2, 'backup_label2', '012 -> 013');
+wait_recovery_and_switch_wal_primary('012 -> 013');
+
+$primary->stop('smart');
+restore_backup_primary($basebackupdir2, 'backup_label2', '012 -> 023');
+wait_recovery_and_switch_wal_primary('012 -> 023');
+
+$primary->stop('smart');
+restore_backup_primary($basebackupdir1, 'backup_label1', '101 -> 112');
+wait_recovery_and_switch_wal_primary('101 -> 112');
+
+$primary->stop('smart');
+restore_backup_primary($basebackupdir2, 'backup_label2', '112 -> 113');
+wait_recovery_and_switch_wal_primary('112 -> 113');
+
+$primary->stop('smart');
+restore_backup_primary($basebackupdir2, 'backup_label2', '112 -> 123');
+wait_recovery_and_switch_wal_primary('112 -> 123');
+
+$primary->stop('smart');
+restore_backup_primary($basebackupdir1, 'backup_label1', '201 -> 212');
+wait_recovery_and_switch_wal_primary('201 -> 212');
+
+$primary->stop('smart');
+restore_backup_primary($basebackupdir2, 'backup_label2', '212 -> 213');
+wait_recovery_and_switch_wal_primary('212 -> 213');
+
+$primary->stop('smart');
+restore_backup_primary($basebackupdir2, 'backup_label2', '212 -> 223');
+wait_recovery_and_switch_wal_primary('212 -> 223');
+
+$primary->teardown_node;
```

Comment:
Could you please remove this line and also move the test to `src/test/recovery` because the change in code is not directly related to `pg_basebackup`?

Reply:
Done.